### PR TITLE
ACC-2127:Change url of livechat depending on the flag liveChatStaging

### PR DESCRIPTION
**Pull request by danieleruiz**
The Salesforce team want to test some changes they are making to the live chat in their staging environment, so they’d like the option to toggle using their staging or production environment.
[ACC-2127](https://financialtimes.atlassian.net/browse/ACC-2127)

[ACC-2127]: https://financialtimes.atlassian.net/browse/ACC-2127?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/js/browser.ts [hunk 1]
```diff
@@ -73,7 +79,9 @@ class LiveChat {
 						}
 						this.button.onclick = () => {
 							if (chatterBox) {
-								const url: string = `${liveChatURL}/${this.config.buttonReference}/${this.config.deploymentId}`;
+								// @ts-ignore
+								const baseUrl: string = this.flags.get('liveChatStaging') ? `https://ip-chatterbox-client-staging.herokuapp.com/` : `https://live-chat.ft.com/`;
```
Comment (asugar13) [inline src/js/browser.ts:83]: **Note**: Can you use environment variables from Vault for the URLs? LIVE_CHAT_STAGING_HOST and LIVE_CHAT_PROD_HOST should be the ones you are looking for

File: src/js/browser.ts [hunk 2]
```diff
@@ -73,7 +79,9 @@ class LiveChat {
 						}
 						this.button.onclick = () => {
 							if (chatterBox) {
-								const url: string = `${liveChatURL}/${this.config.buttonReference}/${this.config.deploymentId}`;
+								// @ts-ignore
```
Comment (asugar13) [inline src/js/browser.ts:82]: **Question**: Why do you need a ts-ignore here?

File: src/js/browser.ts [hunk 3]
```diff
@@ -31,13 +33,17 @@ class LiveChat {
...
 	offlineIndicator: HTMLElement;
+	flags: TFlagsData;
 
 	constructor() {
 		this.container = document.getElementById('liveAgent') as HTMLDivElement;
 		this.button = document.getElementById('liveAgentButton') as HTMLButtonElement;
 		this.onlineIndicator = document.getElementById('liveAgentOnlineIndicator') as HTMLDivElement;
 		this.offlineIndicator = document.getElementById('liveAgentOfflineIndicator') as HTMLDivElement;
 		this.config = this.container.dataset as SalesforceConfig;
+		// @ts-ignore
```
Comment (asugar13) [inline src/js/browser.ts:44]: **Question:**Why do you need a ts-ignore here?

File: tsconfig.json [hunk 4]
```diff
@@ -2,10 +2,10 @@
   "exclude": ["node_modules", "demos"],
   "compilerOptions": {
     "target": "es2017",
+    "jsx": "react",
     "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */                     /* Concatenate and emit output to single file. */
     "outDir": "./dist",                       /* Redirect output structure to the directory. */
     "rootDir": "./src/js",                    /* Specify the root directory of input files. Use to control the output directory structure with --outDir. */
     "removeComments": true,                   /* Do not emit comments to output. */
-    "strict": true		                      /* Enable all strict type-checking options. */
```
Comment (asugar13) [inline tsconfig.json:9]: **Question**: Why was this removed?

File: src/js/browser.ts (hunk 1 above)
Comment (danieleruiz) [reply src/js/browser.ts:83]: thanks Asier! modified

File: src/js/browser.ts (hunk 2 above)
Comment (danieleruiz) [reply src/js/browser.ts:82]: typescript don´t know that pageKitFlags have init method. i did the same as here: [link](https://github.com/Financial-Times/next-home-page/blob/577319b85e171eea2a82e9119d6d0cadc69ea523/apps/home-page/client/setup.ts#L26)

File: src/js/browser.ts (hunk 3 above)
Comment (danieleruiz) [reply src/js/browser.ts:44]: the same as above, typescript doesn´t know that pagekitflags has that method. even if i indicate the type

File: tsconfig.json (hunk 4 above)
Comment (danieleruiz) [reply tsconfig.json:9]: strict mode in compiler makes an error with the new library of dotcom-ui-flags

Comment (asugar13) [reply tsconfig.json:9]: What's the error? Is there a way to get rid of the error using something like ts-ignore instead of changing the linting rules?

Comment (asugar13) [reply tsconfig.json:9]: here are the docs that explain the issue: https://github.com/microsoft/TypeScript/issues/44205
strict checking cannot be enabled only for project files

**Review by asugar13 — APPROVED** [review]
thank you! looks good
